plugins/catalog-backend/api-report.md

```diff
@@ -872,7 +872,7 @@ export type EntityAncestryResponse = {
 // @public
 export type EntityFilter = {
   anyOf: {
-    allOf: EntitiesSearchFilter[];
+    allOf: (EntitiesSearchFilter | EntityFilter)[];
```

Comment:
So this is actually interesting. Makes me think, is there one MORE step we could take to get even farther?

```ts
export type EntityFilter =
  | { allOf: EntityFilter[] }
  | { anyOf: EntityFilter[] }
  | EntitiesSearchFilter;
```

this would address one of my gripes with this structure - you aren't _forced_ to have the whole anyOf/allOf nesting just to get at the single little filter you actually wanted to add.

This should also be backwards compatible. Does that sound like it makes sense? Ping @mtlewis here too

Reply:
Ah, we actually discussed this ourselves when starting this work, but we decided that it might be better to insert the nested functionality first, and then expand into this more free form API afterwards. But if you'd like to see it in the same PR I'm happy to tackle it now - I'll defer to you!

Reply:
Turns out the additional step was pretty straightforward - I've added a new commit for the flexible support.